src/hash.cpp

```diff
@@ -181,7 +181,10 @@ bcrypt bcrypt::hash_pw(const std::string& password, bcrypt& salt)
 }
 
 bool bcrypt::is_valid_prefix(const std::string& hash) {
-	return hash.compare(0, 4, "$2y$") == 0;
+	return ((hash.compare(0, 4, "$2a$") == 0)
+	     || (hash.compare(0, 4, "$2b$") == 0)
+	     || (hash.compare(0, 4, "$2x$") == 0)
+	     || (hash.compare(0, 4, "$2y$") == 0));
```

Comment:
I'd prefer separating the prefix with `hash.substr()`, then comparing it with known prefixes with the `==` operator.

Reply:
That makes a copy (even if it's just 5 bytes + overhead), this does not. You could use a string_view to avoid copying, but this is less work.

Reply:
won't fix